lib/rubocop/cop/rails/rescue_from_exceptions_variable_name.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop makes sure that rescued exceptions variables are named as
+      # expected.
+      #
+      # The `PreferredName` config option takes a `String`. It represents
+      # the required name of the variable. Its default is `e` that is read
+      # from `Naming/RescuedExceptionsVariableName` cop in the main Rubocop
+      # repository.
+      #
+      # @example PreferredName: e (default)
+      #   # bad
+      #   rescue_from MyException do |exception|
+      #     # do something
+      #   end
+      #
+      #   # good
+      #   rescue_from MyException do |e|
+      #     # do something
+      #   end
+      #
+      #   # good
+      #   rescue_from MyException do |_e|
+      #     # do something
+      #   end
+      #
+      # @example PreferredName: exception
+      #   # bad
+      #   rescue_from MyException do |e|
+      #     # do something
+      #   end
+      #
+      #   # good
+      #   rescue_from MyException do |exception|
+      #     # do something
+      #   end
+      #
+      #   # good
+      #   rescue_from MyException do |_exception|
+      #     # do something
+      #   end
+      #
+      class RescueFromExceptionsVariableName < Cop
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Use `%<preferred>s` instead of `%<bad>s`.'
+
+        def on_block(node)
+          name = variable_name(node)
+          return unless name
+          return if preferred_name(name).to_sym == name
+
+          add_offense(node, location: offense_range(node))
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            offending_name = variable_name(node)
+            preferred_name = preferred_name(offending_name)
+            corrector.replace(offense_range(node), "|#{preferred_name}|")
+          end
+        end
+
+        private
+
+        def offense_range(resbody)
```

Comment:
It feels like this would better be expressed with node pattern. Do you need any guidance with that?
Docs: https://docs.rubocop.org/rubocop-ast/node_pattern.html